probe-rs/src/architecture/arm/traits.rs

```diff
@@ -16,6 +16,8 @@ pub enum PortType {
 /// Almost everything is the responsibility of the caller. For example, the caller must
 /// handle bank switching and AP selection.
 pub trait RawDapAccess {
+    fn select_dp(dp: Option<u32>) -> Result<(), DebugProbeError>;
```

Comment:
This would perform the whole line reset sequence, right?

If we want to implement the sequences as specified by ARM, the behaviour for this is implemented by the ` DebugPortSetup` sequence. We will have to call the sequence in the implementation, just something to consider.

Reply:
Oh no.. this will be fun